# The Question

You are given the following information, but you may prefer to do some research for yourself.

1 Jan 1900 was a Monday.
Thirty days has September,
April, June and November.
All the rest have thirty-one,
Saving February alone,
Which has twenty-eight, rain or shine.
And on leap years, twenty-nine.
A leap year occurs on any year evenly divisible by 4, but not on a century unless it is divisible by 400.

How many Sundays fell on the first of the month during the twentieth century (1 Jan 1901 to 31 Dec 2000)?

# Solution  

## Approach

    1) Estimate the solution arithmetically
    
    2) Calculate an exact solution algerythmically and check that it agrees with the estimate  

## Estimate

In [80]:
(12 * 100) / 7 #There are 100 years in a century, and 12 months in a year about one seventh of which begin on a Sunday

171.42857142857142

We would expect there to be about 171 Sundays falling on the first of a month per century. 

## Exact Solution

Calculate an exact solution by iterating through a python, pandas, daterange object counting Sundays that fall on the first of the month.

In [76]:
import pandas as pd

In [77]:
start = dt.date(1901, 1,1)                             # inclusive start date
end = dt.date(2001, 1, 1)                              # exclusive end date
twentieth_century = pd.date_range(start, end)

In [78]:
print(twentieth_century)

DatetimeIndex(['1901-01-01', '1901-01-02', '1901-01-03', '1901-01-04',
               '1901-01-05', '1901-01-06', '1901-01-07', '1901-01-08',
               '1901-01-09', '1901-01-10',
               ...
               '2000-12-23', '2000-12-24', '2000-12-25', '2000-12-26',
               '2000-12-27', '2000-12-28', '2000-12-29', '2000-12-30',
               '2000-12-31', '2001-01-01'],
              dtype='datetime64[ns]', length=36526, freq='D')


In [79]:
sunday_count = 0
for single_day in twentieth_century:
    if single_day.weekday() == 6 and single_day.day == 1:
        sunday_count += 1
print(sunday_count)

171


# Conclusion

One hundred seventy-one Sundays fell on the first of the month during the 20th century. This agrees with the estimate. 